dali/base/dadfs.cpp

```diff
@@ -6627,19 +6627,28 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
 
     void fillClustersCache()
     {
-        if (clusterscache.ordinality()==0) {
+        if (clusterscache.ordinality()==0)
+        {
             StringBuffer name;
-            ForEachItemIn(i,subfiles) {
+            ForEachItemIn(i,subfiles)
+            {
                 StringArray clusters;
                 IDistributedFile &f=subfiles.item(i);
                 Owned<IFileDescriptor> fdesc = f.getFileDescriptor();
+                StringArray clusterNames;
+                f.getClusterNames(clusterNames);
                 unsigned nc = f.numClusters();
-                for(unsigned j=0;j<nc;j++) {
-                    f.getClusterName(j,name.clear());
+                for(unsigned j=0;j<nc;j++)
+                {
+                    name.set(clusterNames.item(j));
```

Comment:
minor: Could you use
const char * name = clusterNames.item(j);
I don't think you need to copy it to a StringBuffer.

Reply:
@ghalliday - I've squashed in that change.